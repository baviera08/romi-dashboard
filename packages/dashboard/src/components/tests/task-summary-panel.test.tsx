import * as RomiCore from '@osrf/romi-js-core-interfaces';
import React from 'react';
import TreeView from '@material-ui/lab/TreeView';
import TreeItem from '@material-ui/lab/TreeItem';
import TaskSummaryPanel from '../task-summary-panel';
import fakeTaskSummary from '../../mock/data/task-summary';
import { createMount } from '@material-ui/core/test-utils';
import { shallow } from 'enzyme';

const mount = createMount();

test('Matches snapshot', () => {
  const tasks = Object.values(fakeTaskSummary());
  const root = shallow(<TaskSummaryPanel allTasks={tasks} />);
  expect(root).toMatchSnapshot();
  root.unmount();
});

test('Renders tree items', () => {
  const tasks = Object.values(fakeTaskSummary());
  const root = mount(<TaskSummaryPanel allTasks={tasks} />);
  expect(root.find(TreeView).exists()).toBeTruthy();
  expect(root.find(TreeItem).length).toBe(tasks.length);
  root.unmount();
});

test('Show description below the id if the task has an actor', () => {
  const tasks = Object.values(fakeTaskSummary());
  const root = mount(<TaskSummaryPanel allTasks={tasks} />);
  expect(root.html()).toContain('makeStyles-taskActor');
  root.unmount();
});

test('Does not show description below the id if the task has an actor', () => {
  const task = Object.values(fakeTaskSummary())[0];
  task.status = 'Finished';
  const root = mount(<TaskSummaryPanel allTasks={[task]} />);
  expect(root.html().includes('makeStyles-taskActor')).toBeFalsy();
  root.unmount();
});

describe('Components gets the correct style on specifics states', () => {
  let task: RomiCore.TaskSummary;
  beforeEach(() => {
    task = Object.values(fakeTaskSummary())[0];
  });

  test('active style is applied ', () => {
    task.state = RomiCore.TaskSummary.STATE_ACTIVE;
    const root = mount(<TaskSummaryPanel allTasks={[task]} />);
    expect(root.find(TreeItem).html()).toContain('makeStyles-active');
    root.unmount();
  });

  test('queue style is applied', () => {
    task.state = RomiCore.TaskSummary.STATE_QUEUED;
    const root = mount(<TaskSummaryPanel allTasks={[task]} />);
    expect(root.find(TreeItem).html()).toContain('makeStyles-queued');
    root.unmount();
  });

  test('completed style is applied', () => {
    task.state = RomiCore.TaskSummary.STATE_COMPLETED;
    const root = mount(<TaskSummaryPanel allTasks={[task]} />);
    expect(root.find(TreeItem).html()).toContain('makeStyles-completed');
    root.unmount();
  });

  test('failed style is applied', () => {
    task.state = RomiCore.TaskSummary.STATE_FAILED;
    const root = mount(<TaskSummaryPanel allTasks={[task]} />);
    expect(root.find(TreeItem).html()).toContain('makeStyles-failed');
    root.unmount();
  });
});
