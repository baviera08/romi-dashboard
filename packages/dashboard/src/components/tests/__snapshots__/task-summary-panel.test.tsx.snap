// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<WithStyles(ForwardRef(TreeView))
  className="makeStyles-root-1"
  defaultCollapseIcon={<UNDEFINED />}
  defaultExpandIcon={<UNDEFINED />}
  defaultExpanded={
    Array [
      "root",
    ]
  }
  expanded={Array []}
  onNodeSelect={[Function]}
  onNodeToggle={[Function]}
  selected={Array []}
>
  <WithStyles(ForwardRef(TreeItem))
    classes={
      Object {
        "expanded": "makeStyles-expanded-5",
        "label": "makeStyles-active-8 makeStyles-labelContent-4",
        "root": "makeStyles-treeChildren-3",
      }
    }
    data-component="TreeItem"
    key="af8faee9-84ca-41ea-8bb6-8493cc9f824c"
    label={
      <React.Fragment>
        <ForwardRef(WithStyles)
          noWrap={true}
          variant="body1"
        >
          <ForwardRef(WithStyles)
            onClick={[Function]}
          >
            <UNDEFINED />
          </ForwardRef(WithStyles)>
          af8faee9-84ca-41ea-8bb6-8493cc9f824c
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          className="makeStyles-taskActor-10"
          variant="body1"
        >
          [tinyRobot/tinyRobot1]
        </ForwardRef(WithStyles)>
      </React.Fragment>
    }
    nodeId="af8faee9-84ca-41ea-8bb6-8493cc9f824c"
  >
    <TaskSummaryPanelItem
      key="af8faee9-84ca-41ea-8bb6-8493cc9f824c"
      task={
        Object {
          "end_time": Object {
            "nanosec": 0,
            "sec": 0,
          },
          "start_time": Object {
            "nanosec": 0,
            "sec": 0,
          },
          "state": 1,
          "status": "Moving [tinyRobot/tinyRobot1]: ( 9.81228 -6.98942 -3.12904) -> ( 6.26403 -3.51569  1.16864) | Remaining phases: 1 | Remaining phases: 6",
          "submission_time": Object {
            "nanosec": 0,
            "sec": 0,
          },
          "task_id": "af8faee9-84ca-41ea-8bb6-8493cc9f824c",
        }
      }
    />
  </WithStyles(ForwardRef(TreeItem))>
  <WithStyles(ForwardRef(TreeItem))
    classes={
      Object {
        "expanded": "makeStyles-expanded-5",
        "label": "makeStyles-active-8 makeStyles-labelContent-4",
        "root": "makeStyles-treeChildren-3",
      }
    }
    data-component="TreeItem"
    key="am8faee9-84ca-41ea-8bb6-8493cc9f8249"
    label={
      <React.Fragment>
        <ForwardRef(WithStyles)
          noWrap={true}
          variant="body1"
        >
          <ForwardRef(WithStyles)
            onClick={[Function]}
          >
            <UNDEFINED />
          </ForwardRef(WithStyles)>
          am8faee9-84ca-41ea-8bb6-8493cc9f8249
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          className="makeStyles-taskActor-10"
          variant="body1"
        >
          [tinyRobot/tinyRobot2]
        </ForwardRef(WithStyles)>
      </React.Fragment>
    }
    nodeId="am8faee9-84ca-41ea-8bb6-8493cc9f8249"
  >
    <TaskSummaryPanelItem
      key="am8faee9-84ca-41ea-8bb6-8493cc9f8249"
      task={
        Object {
          "end_time": Object {
            "nanosec": 0,
            "sec": 0,
          },
          "start_time": Object {
            "nanosec": 0,
            "sec": 0,
          },
          "state": 1,
          "status": "Moving [tinyRobot/tinyRobot2]: ( 9.81228 -6.98942 -3.12904) -> ( 6.26403 -3.51569  1.16864) | Remaining phases: 1 | Remaining phases: 6",
          "submission_time": Object {
            "nanosec": 0,
            "sec": 0,
          },
          "task_id": "am8faee9-84ca-41ea-8bb6-8493cc9f8249",
        }
      }
    />
  </WithStyles(ForwardRef(TreeItem))>
</WithStyles(ForwardRef(TreeView))>
`;
